custom-assembly-report-generator/generate_custom_assembly_report.py

```diff
@@ -0,0 +1,184 @@
+    return pandas.concat([chunk[chunk['#rs_assembly'] == assembly_accession] for chunk in iter_csv])
+
+
+def get_refseq_accessions_without_equivalent_genbank_accessions(assembly_report_dataframe):
+    return set(assembly_report_dataframe[((assembly_report_dataframe["Relationship"] == "<>") |
+                                         (assembly_report_dataframe["GenBank-Accn"] == "na")) &
+                                         (assembly_report_dataframe["RefSeq-Accn"] != "na")]["RefSeq-Accn"]
+               .values.tolist())
+
+
+def get_genbank_accessions_without_equivalent_refseq_accessions(assembly_report_dataframe):
+    return set(assembly_report_dataframe[((assembly_report_dataframe["Relationship"] == "<>") |
+                                         (assembly_report_dataframe["RefSeq-Accn"] == "na")) &
+                                         (assembly_report_dataframe["GenBank-Accn"] != "na")]["GenBank-Accn"]
+               .values.tolist())
+
+
+def get_absent_refseq_accessions(species_info, refseq_accessions_from_assembly_report):
+    pg_conn = data_ops.get_pg_conn_for_species(species_info)
+    pg_cursor = pg_conn.cursor()
+    contiginfo_tables = data_ops.get_contiginfo_table_list_for_schema(pg_cursor, species_info["database_name"])
+    # Can't filter by assembly accession because only some contiginfo tables have that field :(
+    contiginfo_query = "select distinct * from ({0}) temp where contig_name not in ({1})".format(
+        " union all ".join(["select contig_name, coalesce(contig_chr, contig_name) as chromosome, "
+                            "group_term as assembly_unit, contig_length from dbsnp_{0}.{1}"
+                           .format(species_info["database_name"], contiginfo_table_name)
+                            for contiginfo_table_name in contiginfo_tables]),
+        ','.join(["'{0}'".format(elem) for elem in refseq_accessions_from_assembly_report])
+    )
+    pg_cursor.execute(contiginfo_query)
+    accessions = pg_cursor.fetchall()
+    pg_cursor.close()
+    pg_conn.close()
+    return accessions
+
+
+def download_assembly_report(assembly_accession):
+    assembly_report_url = get_assembly_report_url(assembly_accession)
+    assembly_report_path = os.getcwd() + os.path.sep + os.path.basename(assembly_report_url)
+    urllib.urlretrieve(assembly_report_url, assembly_report_path)
+    urllib.urlcleanup()
+    return assembly_report_path
+
+
+def insert_absent_genbank_accessions_in_assembly_report(species_db_info, assembly_report_dataframe,
+                                                        genbank_equivalents_dataframe):
+    absent_refseq_accessions = \
+        get_absent_refseq_accessions(species_db_info, set(assembly_report_dataframe["RefSeq-Accn"]))
+    if len(absent_refseq_accessions) == 0:
+        print("INFO: No GenBank accessions are absent in the assembly report for " + species_db_info["database_name"])
+    new_assembly_report_entry_index = len(assembly_report_dataframe)

```

Comment:
Blank line to increase readability, otherwise it is too much text.

```suggestion

    new_assembly_report_entry_index = len(assembly_report_dataframe)
```